### PR TITLE
376 new operator approval

### PR DESCRIPTION
**Pull request by BCerki**
https://github.com/bcgov/cas-registration/issues/376

Notes:
- there are comments in the ticket about approach and scope
- re this part of the AC: 'the internal user will have to explicitly Approve the Operator before the prime admin request can be approved.' This PR allows separate approval of the operator, which is on the first step of the form, before the admin request approval. However, this PR doesn't stop an IRC user from continuing on if they reject the operator. Whatever that will look like will be handled here: https://github.com/bcgov/cas-registration/issues/474

### CODE REVIEW

File: bc_obps/registration/api/operator.py
Comment (BCerki) [inline bc_obps/registration/api/operator.py:96]: Re this AC: if the internal user Requests Changes for the new Operator, the is_new flag will remain as True until the Operator's status is changed to Approved

`is_new` defaults to true, so we don't have to do anything besides not include the CHANGES status to implement this AC

File: bc_obps/registration/models.py
Comment (BCerki) [inline bc_obps/registration/models.py:347]: Is this clear enough?

File: client/app/components/form/UserOperatorMultiStepForm.tsx
Comment (Sepehr-Sobhani) [inline client/app/components/form/UserOperatorMultiStepForm.tsx:103]: nit: suggest using `operator_id` rather than `id` as it would be more meaningful.

File: client/app/components/form/UserOperatorMultiStepForm.tsx
Comment (Sepehr-Sobhani) [inline client/app/components/form/UserOperatorMultiStepForm.tsx:98]: Just to confirm, should we only display review buttons to approve an admin request on the second page?

**Review by Sepehr-Sobhani — COMMENTED** [review]
Nice work! 
I have a couple of suggestions and a question. Additionally, I'd suggest implementing a check on the useroperator's `update-status` endpoints to prevent updating the status of a useroperator if `is_new` is set to True. (just an extra check)

File: client/app/components/form/UserOperatorMultiStepForm.tsx
Comment (BCerki) [reply client/app/components/form/UserOperatorMultiStepForm.tsx:98]: If it's a existing operator, yes, we should only display the buttons on the second page.

File: bc_obps/registration/api/operator.py
Comment (BCerki) [reply bc_obps/registration/api/operator.py:99]: changing it caused this: django.core.exceptions.ValidationError: ['“User object (4da70f32-65fd-4137-87c1-111f2daba3dd)” is not a valid UUID.']

**Comment by BCerki** [comment]
> Nice work! I have a couple of suggestions and a question. Additionally, I'd suggest implementing a check on the useroperator's `update-status` endpoints to prevent updating the status of a useroperator if `is_new` is set to True. (just an extra check)

I'm going to add this to the ticket about handling a rejected operator

File: client/app/components/button/Review.tsx
Comment (andrea-williams) [inline client/app/components/button/Review.tsx:49]: delete?

File: client/app/components/button/Review.tsx
Comment (andrea-williams) [inline client/app/components/button/Review.tsx:93]: delete?

File: client/app/components/button/Review.tsx
Comment (andrea-williams) [inline client/app/components/button/Review.tsx:99]: delete?

File: client/app/components/routes/access-requests/form/UserOperatorReview.tsx
Comment (andrea-williams) [inline client/app/components/routes/access-requests/form/UserOperatorReview.tsx:21]: 🤭
delete please

**Review by andrea-williams — COMMENTED** [review]
Aside from some surplus console logs, code looks good. Will pull and run the code locally now

**Comment by BCerki** [comment]
Note to B: check the mock data--senior officer is missing for user operator 4

**Comment by andrea-williams** [comment]
Ran the code locally - confirmed that internal users can now approve an Operator and can independently approve/reject a prime admin request. 

Some notes for discussion next week (probably separate tickets outside the scope of this one):

- the first section of the Operator form has Approve/Reject buttons that appear when it's a new Operator that needs to be reviewed. IMO it's not super clear that when an internal user is clicking those buttons, they're approving the Operator. Similarly, the second section of the form has Approve/Reject buttons for reviewing the prime admin access request, but it's especially unclear in that case because the form only shows information about the Senior Officer for the Operator, who might be different from the user requesting prime admin. There's no information about the user requesting prime admin shown on that page, it's only shown in the table of all Operators. I think we should be doing something like adding a third section that displays info about the user requesting prime admin and have the Review component there.
- there doesn't appear to be any checks on the backend for missing fields on the Senior Officer data - I was able to "approve" even though the data was empty.
- when the Operator is_new=True, our UI says that internal staff must approve the operator before they can approve the prime admin request, but actually an internal user can click the "Next" button in the form and approve/reject prime admin without approving the new Operator. We really shouldn't allow this - if an operator has an assigned admin, we need the operator to be approved.

File: client/app/components/routes/access-requests/form/UserOperatorReview.tsx
Comment (BCerki) [reply client/app/components/routes/access-requests/form/UserOperatorReview.tsx:21]: (sorry about this)

**Review by andrea-williams — APPROVED** [review]

**Pull request merged**
